ctl/accesslog/accesslog.go

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package accesslog
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"kmesh.net/kmesh/ctl/utils"
+	"kmesh.net/kmesh/pkg/logger"
+)
+
+const (
+	patternAccesslog = "/accesslog"
+)
+
+var log = logger.NewLoggerScope("kmeshctl/accesslog")
+
+func NewCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "accesslog",
+		Short:   "enable or disable Kmesh's accesslog",
+		Example: `# enable Kmesh's accesslog: kmeshctl accesslog <kmesh-daemon-pod> enable`,
+		Args:    cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			SetAccesslog(cmd, args)
+		},
+	}
+	return cmd
+}
+
+func SetAccesslog(cmd *cobra.Command, args []string) {
+	podName := args[0]
+	accesslogFlag := args[1]
+	if accesslogFlag != "true" && accesslogFlag != "false" {
+		log.Errorf("Error: Argument must be 'true' or 'false'")
+		os.Exit(1)
+	}
+
+	fw, err := utils.CreateKmeshPortForwarder(podName)
+	if err != nil {
+		log.Errorf("failed to create port forwarder for Kmesh daemon pod %s: %v", podName, err)
+		os.Exit(1)
+	}
+	if err := fw.Start(); err != nil {
+		log.Errorf("failed to start port forwarder for Kmesh daemon pod %s: %v", podName, err)
+		os.Exit(1)
+	}
+	defer fw.Close()
+
+	data, err := json.Marshal(accesslogFlag)
+	if err != nil {
+		log.Errorf("Error marshaling accesslog info: %v", err)
+		return
+	}
+
+	url := fmt.Sprintf("http://%s%s", fw.Address(), patternAccesslog)
+
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(data))
```

Comment:
If there is only one bool flag, can it be placed directly in the URL, like `http://../accesslog?enable=true`?

Putting it in the body is more cubersome.

Reply:
Done